runtime/sema/errors.go

```diff
@@ -1390,11 +1399,22 @@ func (n MemberMismatchNote) Message() string {
 	return "mismatch here"
 }
 
+// NestedConformanceMismatchNote
+
+type NestedConformanceMismatchNote struct {
+	nestedInterfaceType *InterfaceType
+	ast.Range
+}
+
+func (n NestedConformanceMismatchNote) Message() string {
+	return fmt.Sprintf("does not conform to nested interface requirement `%s`", n.nestedInterfaceType)
+}
+
 // DuplicateConformanceError
 //
 // TODO: just make this a warning?
 type DuplicateConformanceError struct {
-	CompositeType *CompositeType
+	CompositeType CompositeKindedType
```

Comment:
Maybe rename `CompositeType` to `CompositeKindedType`, because it may now also be an `InterfaceType`.
Same for the other errors below